tests/tests_integration/test_api/test_simulators/test_simulators.py

```diff
@@ -0,0 +1,290 @@
+import random
+import time
+import uuid
+
+import pytest
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes.files import FileMetadata
+from cognite.client.data_classes.simulators.filters import (
+    SimulatorIntegrationFilter,
+    SimulatorModelRevisionsFilter,
+    SimulatorModelsFilter,
+    SimulatorRoutineRevisionsFilter,
+    SimulatorRoutinesFilter,
+)
+from cognite.client.data_classes.simulators.simulators import CreatedTimeSort
+from cognite.client.exceptions import CogniteAPIError
+from tests.tests_integration.test_api.test_simulators.seed.data import (
+    data_set_id,
+    resource_names,
+    simulator,
+    simulator_integration,
+    simulator_model,
+    simulator_model_revision,
+    simulator_routine,
+    simulator_routine_revision,
+)
+
+
+def truncated_uuid4():
+    return str(uuid.uuid4())[:8]
+
+
+model_unique_external_id = f"{resource_names['simulator_model_external_id']}_{truncated_uuid4()}"
+model_revision_unique_external_id = f"{resource_names['simulator_model_revision_external_id']}_{truncated_uuid4()}"
+simulator_routine_unique_external_id = f"{resource_names['simulator_routine_external_id']}_{truncated_uuid4()}"
+
+
+@pytest.fixture(scope="class")
+def seed_resource_names() -> dict[str, str]:
+    return resource_names
+
+
+@pytest.fixture
+def seed_file(cognite_client: CogniteClient, seed_resource_names) -> FileMetadata | None:
+    # check if file already exists
+    file = cognite_client.files.retrieve(external_id=seed_resource_names["simulator_model_file_external_id"])
+    if (file is None) or (file is False):
+        file = cognite_client.files.upload(
+            path="tests/tests_integration/test_api/test_simulators/seed/ShowerMixer.dwxmz",
+            external_id=seed_resource_names["simulator_model_file_external_id"],
+            name="ShowerMixer.dwxmz",
+            data_set_id=data_set_id,
+        )
+    yield file
+
+
+@pytest.fixture
+def seed_simulator(cognite_client: CogniteClient, seed_resource_names) -> None:
+    simulator_external_id = seed_resource_names["simulator_external_id"]
+    simulators = cognite_client.simulators.list()
+    simulator_exists = len(list(filter(lambda x: x.external_id == simulator_external_id, simulators))) > 0
```

Comment:
Is there no retrieve you can use? If you only have `list` you can do:
```
simulator_exists = simulators.get(external_id=simulator_external_id)
```
which is a O(1) lookup instead of O(N)